features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/ImageWireframeHelper.kt

```diff
@@ -65,7 +76,55 @@ internal class ImageWireframeHelper(
         return imageWireframe
     }
 
-    private companion object {
-        private const val DRAWABLE_CHILD_NAME = "drawable"
+    @Suppress("NestedBlockDepth")
+    internal fun createCompoundDrawableWireframes(
+        view: TextView,
+        mappingContext: MappingContext,
+        startIndex: Int
+    ): MutableList<MobileSegment.Wireframe> {
+        val result = mutableListOf<MobileSegment.Wireframe>()
+        var currentIndex = startIndex
+        val density = mappingContext.systemInformation.screenDensity
+
+        view.compoundDrawables.forEachIndexed { index, _ ->
+            val drawable = view.compoundDrawables[index]
+
+            if (drawable != null) {
+                val drawableCoordinates = viewUtilsInternal.resolveCompoundDrawableBounds(
+                    view = view,
+                    drawable = drawable,
+                    pixelsDensity = density,
+                    index = index
+                )
+
+                val drawableX: Long? = drawableCoordinates?.x
+                val drawableY: Long? = drawableCoordinates?.y
+
+                if (drawableX != null && drawableY != null) {
+                    @Suppress("ThreadSafety") // TODO REPLAY-1861 caller thread of .map is unknown?
+                    createImageWireframe(
+                        view = view,
+                        index = ++currentIndex,
```

Comment:
I would also suggest to convert index to enum in this method instead of doing it inside `createImageWireframe`, this has several advantages:

* we don't use `index` in the `createImageWireframe`, it removes confusion because when inside `createImageWireframe` it is not clear what is the index.
* `createImageWireframe` signature is cleaner, it will get an enum.
* it will remove the need for `CompoundDrawablePositions` to have `position` property, because you can create it right in the beginning of the `forEachIndexed` lambda in the following way:

```
val drawablePosition = when(index) {
   0 -> ...
   1 -> ...
   ...
}
```
since it will be close to the comment describing `compoundDrawables` property, it will be clear what is the index.

Reply:
I understand the confusion here - this is a different index. The index for compoundDrawable represents the left, top, right, bottom drawable, however the index for createImageWireframe represents the index of the wireframe in the stack. Since we are creating potentially multiple imagewireframes off the same view we need a way to generate unique ids for each wireframe to avoid collisions - this can be seen in line39 of imageWireframeHelper where the index is appended to the resolveChildUniqueIdentifier call.

Reply:
Then still maybe it is better to remove this confusion, because here https://github.com/DataDog/dd-sdk-android/pull/1592/files#diff-7870c4af39004135266ff0b27eb0e6ee96774f0443572f43f1225b716dc9a0d3R80 it is still 0 to left, etc. So we can create enum in the current method, and let then index to be used in `createImageWireframe` only for ID generation.